src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -341,72 +343,97 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                         }
 
-                        // Check if this is a rename
-                        if (IsFlagSet(eventFlags[i], Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRenamed))
+                        var changedFlags = FSEventStreamEventFlags.kFSEventStreamEventFlagItemInodeMetaMod |
+                                           FSEventStreamEventFlags.kFSEventStreamEventFlagItemFinderInfoMod |
+                                           FSEventStreamEventFlags.kFSEventStreamEventFlagItemModified |
+                                           FSEventStreamEventFlags.kFSEventStreamEventFlagItemChangeOwner |
+                                           FSEventStreamEventFlags.kFSEventStreamEventFlagItemXattrMod;
+
+                        // If any of the Changed flags are set in both Filter and Event, return true as a Changed event has occurred.
+                        if (((_filterFlags & changedFlags) & (eventFlags[i] & changedFlags)) > 0)
                         {
-                            // Find the rename that is paired to this rename, which should be the next rename in the list
-                            long pairedId = FindRenameChangePairedChange(i, eventPaths, eventFlags, eventIds);
-                            if (pairedId == long.MinValue)
-                            {
-                                // Getting here means we have a rename without a pair, meaning it should be a create for the 
-                                // move from unwatched folder to watcher folder scenario or a move from the watcher folder out.
-                                // Check if the item exists on disk to check which it is
-                                WatcherChangeTypes wct = DoesItemExist(eventPaths[i], IsFlagSet(eventFlags[i], Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemIsFile)) ?
-                                    WatcherChangeTypes.Created : 
-                                    WatcherChangeTypes.Deleted;
-                                watcher.NotifyFileSystemEventArgs(wct, relativePath);
-                            }
-                            else
-                            {
-                                // Remove the base directory prefix and add the paired event to the list of 
-                                // events to skip and notify the user of the rename 
-                                string newPathRelativeName = eventPaths[pairedId].Remove(0, _fullDirectory.Length);
-                                watcher.NotifyRenameEventArgs(WatcherChangeTypes.Renamed, newPathRelativeName, relativePath);
+                            watcher.NotifyFileSystemEventArgs(WatcherChangeTypes.Changed, relativePath);
+                        }
+
+                        // Disallow event coalescing. OSX likes to add a Created event to everything, particularly Changed events.
```

Comment:
If an actual created event has occurred, it will happen in a separate event that only has ```Created``` and the file entry type (File, Directory, SymLink, HardLink, LastHardLink). I added some tests for this to be sure we weren't throwing away legitimate events.